### DISCUSSION TITLE
Allow hide required property with another required property CS9031

### DISCUSSION THREAD

**Discussion by UniqeId6542** [dotnet/csharplang#6872] [General]
If a 'new' property hides and sets the base class's property, it should not make a compiler error.  (Or, add an attribute to help support it)

```
public class A
{
    public required object Value { get; init; }
}

public class B : A
{
    public new required string Value // CS9031 - should compile as base.Value is set in init
    {
        get => (string)base.Value;
        init => base.Value = value;
    }
}
```

**Comment by RikkiGibson** [comment]
This is a language design suggestion, so moving to csharplang.

**Comment by TahirAhmadov** [comment]
The problem here is that we would need some syntax to "advertise" that `B.Value` unconditionally sets `A.Value`. I don't think C# typically interprets the code in methods to infer the signature (including property setters). Also, what about properties which have a different name, and not `new` (hiding) ones?

**Reply by UniqeId6542** [reply]
> we would need some syntax to "advertise"

Yes.  Constructors can use the `SetsRequiredMembers` attribute.  This issue would (probably) need something where you could pass in the property name.  The implementation gets more difficult in needing to have a way to differentiate `B.Value` from `A.Value`.
